tracing-api/src/main/java/com/palantir/tracing/api/OpenSpan.java

```diff
@@ -77,6 +83,13 @@ public static Builder builder() {
                 .startClockNanoSeconds(System.nanoTime());
     }
 
+    /**
+     * Use this factory method to avoid allocate {@link Builder} in hot path.
+     */
+    public static OpenSpan of(String operation, String spanId, SpanType type, Optional<String> parentSpanId) {
```

Comment:
It's a bit of a shame that we have to make this public in order to access it within this library, as I really hope that no users will ever actually touch this class!